third_party/txt/src/txt/paragraph.cc

```diff
@@ -1394,6 +1386,48 @@ void Paragraph::PaintDecorations(SkCanvas* canvas,
   }
 }
 
+void Paragraph::ComputeWavyDecoration(SkPath& path,
+                                      double x,
+                                      double y,
+                                      double width,
+                                      double thickness) {
+  int wave_count = 0;
```

Comment:
Just a thought.  Rather than keep an integer wave_count and test it for oddness to invert the wave, instead use ```double wave_height = quarter;``` and then in each loop you can use ```wave_height = -wave_height;" instead of ```++wave_count;``` combined with the modulus tests...